skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillsController.java

```diff
@@ -0,0 +1,69 @@
+package com.RDS.skilltree.Skill;
+
+import com.RDS.skilltree.Exceptions.NoEntityException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.coyote.Response;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.ObjectUtils;
+import org.springframework.web.bind.annotation.*;
+
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+@RestController
+@Slf4j
+@RequestMapping("/v1/skills")
+public class SkillsController {
+    private final SkillsService skillsService;
+    public SkillsController(SkillsService skillsService){
+        this.skillsService = skillsService;
+    }
+
+    @PostMapping("/")
+    public ResponseEntity<?> createSkill(@RequestBody(required = true) SkillDRO skillDRO){
+        if (skillDRO.getCreatedBy() == null || skillDRO.getType() == null || skillDRO.getName() == null){
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body("CreatedBy, Type and Name are mandatory values, anyone cannot be null");
+        }
```

Comment:
We can use java validation annotation @NotNull instead of manually checking this.

Reply:
I tried using that, in that case we were not able to show custom error message to the user.

Reply:
Did you use @Valid annotation on Request body?
https://www.baeldung.com/spring-boot-bean-validation

Reply:
updated, pushing the commits

Reply:
updated